packages/canary-web-components/src/components/ic-table-of-contents/ic-table-of-contents.tsx

```diff
@@ -0,0 +1,442 @@
+            ? "bottom"
+            : this.position === "right"
+            ? "left"
+            : "right";
+        //Modify the ic-tooltip wrapped around the list item
+        tooltip.setAttribute("disable-click", false);
+        tooltip.setAttribute("disable-hover", false);
+        tooltip.setAttribute("placement", tooltipPosition);
+        tooltip.setAttribute("target", item.id);
+        tooltip.setAttribute("style", "overflow:hidden");
+      } else {
+        //disable the tooltip if the item does not need to truncate
+        tooltip.setAttribute("disable-click", true);
+        tooltip.setAttribute("disable-hover", true);
+      }
+    }
+  }
+
+  // truncate text if it goes out of bounds of the container
+  private truncateText = () => {
+    const links = this.el.shadowRoot?.querySelectorAll(`.${TOC_ITEM_CLASS}`);
+    const contentList = this.el.shadowRoot?.querySelector(
+      `.${TOC_LIST_ITEMS_CLASS}`
+    );
+    if (contentList)
+      links?.forEach((item) => {
+        this.addTooltip(item, DEVICE_SIZES.S);
+      });
+  };
+
+  // initialise resize observer
+  componentWillLoad(): void {
+    this.resizeObserver = new ResizeObserver(() => this.handleResizeBounds());
+    this.resizeObserver.observe(this.el);
+  }
+
+  // initialise scroll listener
+  componentDidRender(): void {
+    document.addEventListener("scroll", () => {
+      if (!this.suppressListener) this.updateSelectedItemFromScroll();
+      else this.suppressListener = false;
+    });
+    this.containerPosition();
+  }
+
+  // after component has loaded modify the list based of position, size and surrounding elements
+  async componentDidLoad(): Promise<void> {
+    this.focusedElement = 0;
+    this.headerHeight = this.findHeaderElement();
+    this.addListMargin();
+    this.truncateText();
+    this.findElementsHeights();
+  }
+
+  // Component is a wrapper for page contents, and will build list from heading items
+  render() {
+    const { heading, monochrome, position, size, theme } = this;
+    return (
+      <Host class={{ [`ic-theme-${theme}`]: theme !== "inherit" }}>
+        <ic-section-container
+          id={"table-of-contents-section-container"}
+          aligned={"full-width"}
+          style={{ display: "flex" }}
+        >
+          <div
+            role="navigation"
+            aria-label="Table of contents"
+            class={`${TOC_CONTENT_CLASS} ${
+              position === "top" && "table-of-contents-position-top"
+            }
+            ${position === "right" && "table-of-contents-position-side"}`}
+          >
+            <ic-typography variant="subtitle-large">{heading}</ic-typography>
+            <div class={"table-of-contents-list"}>
+              <div
+                class={`table-of-contents-sidebar ${
+                  monochrome && `table-of-contents-sidebar-monochrome`
+                }`}
+                style={{ height: size === "large" ? "2rem" : "1.75rem" }}
```

Comment:
Could this styling be moved into the CSS file instead? As we generally don't add inline styling like this in other components

Same with line 412